UIforETW/UIforETWDlg.cpp

```diff
@@ -433,27 +1196,47 @@ BOOL CUIforETWDlg::OnInitDialog()
 	return TRUE; // return TRUE unless you set the focus to a control
 }
 
-std::wstring CUIforETWDlg::GetDirectory(const wchar_t* env, const std::wstring& default)
+void CUIforETWDlg::initializeToolTip( )
+{
+	
+	toolTip_.SetMaxTipWidth(400);
+	toolTip_.Activate(TRUE);
+
+	addSingleToolToToolTip(&toolTip_, &btStartTracing_, StartEtwTracingString);
+	addSingleToolToToolTip(&toolTip_, &btCompress_, btCompressToolTipString);
+	addSingleToolToToolTip(&toolTip_, &btCswitchStacks_, btCswitchStacksString);
+	addSingleToolToToolTip(&toolTip_, &btSampledStacks_, SampledStacksString);
+	addSingleToolToToolTip(&toolTip_, &btFastSampling_, SampleRateString);
+	addSingleToolToToolTip(&toolTip_, &btGPUTracing_, GPUTracingString);
+	addSingleToolToToolTip(&toolTip_, &btShowCommands_, ShowCommandsString);
+	addSingleToolToToolTip(&toolTip_, &btInputTracingLabel_, InputTipString);
+	addSingleToolToToolTip(&toolTip_, &btInputTracing_, InputTipString);
+	addSingleToolToToolTip(&toolTip_, &btTracingMode_, TracingModeString);
+	addSingleToolToToolTip(&toolTip_, &btTraces_, TracesString);
+	addSingleToolToToolTip(&toolTip_, &btTraceNotes_, TraceNotesString);
+}
+
+std::wstring CUIforETWDlg::GetDirectory( _In_z_ PCWSTR env, const std::wstring& default)
 {
 	// Get a directory (from an environment variable, if set) and make sure it exists.
-	std::wstring result = default;
-#pragma warning(suppress : 4996)
-	const wchar_t* traceDir = _wgetenv(env);
-	if (traceDir)
-	{
-		result = traceDir;
-	}
-	// Make sure the name ends with a backslash.
-	if (!result.empty() && result[result.size() - 1] != '\\')
-		result += '\\';
-	if (!PathFileExists(result.c_str()))
+
+	std::wstring result = getTraceDirFromEnvironmentVariable(env, default);
+	const bool doesFileExist = isValidPathToFSObject(result);
+	if (!doesFileExist)
 	{
-		(void)_wmkdir(result.c_str());
+		const bool sucessfullyCreatedDirectory =
+			enhancedCreateDirectory(result);
+		if (!sucessfullyCreatedDirectory)
+		{
+			AfxMessageBox(L"CUIforETWDlg::GetDirectory attempted to create a directory, but unexpectedly failed!");

```

Comment:
Should say what directory failed to be created in order to make the error message useful.